web/src/buddy_list.ts

```diff
@@ -723,8 +723,9 @@ export class BuddyList extends BuddyListConf {
 
         // After the `await`, we might have changed to a different channel view.
         // If so, we shouldn't update the DOM anymore, and should let the newer `populate`
-        // call set things up with fresh data.
-        if (current_sub !== this.render_data.current_sub) {
+        // call set things up with fresh data. We also want to make sure not to show
+        // the links when searching users, which might have changed since fetching data.
+        if (current_sub !== this.render_data.current_sub || buddy_data.get_is_searching_users()) {
```

Comment:
I think ideally this comment should note that in the search case, a different callback is responsible for updating the rendering?

Reply:
I don't think that's true/relevant for user links? In the search case, we don't show the user links. If the search string changes, this will be called again (likely in the same way it was this time) and if search is over then the user links will be shown again (if relevant).

Reply:
OK I guess I really don't explain the logic for why it's correct to return here if we're searching for users, especially given we didn't do so previously? It seems to me that this risks having stale state for the links left in the sidebar.

Reply:
We did do it previously:

```js
        if (!buddy_data.get_is_searching_users()) {
            void this.render_view_user_list_links();
        }
```

If it's easier to follow, I can change this to an early return in `render_view_user_list_links`?

Reply:
Yeah moving that, with a good comment, would be a good follow-up.